### PR TITLE
Revert mitigation for SQ regression on our ITs

### LINKED ISSUE #6714: Revert changes on Java ITs which were made due to SQ regression
Context: https://sonarsource.slack.com/archives/C012KBFFYD6/p1675346806988809?thread_ts=1675342178.975529&cid=C012KBFFYD6

Change `contains` to `containsExactlyInOrder` all checks on `l.contains("WARN")))`

We've reverted them in https://github.com/SonarSource/sonar-dotnet/pull/6715 because of a SQ regression which polluted the warning logs.

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Fixes #6714

### CODE REVIEW

File: its/src/test/java/com/sonar/it/shared/ScannerCliTest.java [hunk 1]
```diff
@@ -102,14 +102,15 @@ public void givenTestHtmlCode_whenScannerForCliIsUsed_doesNotLogCsharpWarning()
...
+    assertThat(result.getLogsLines(l -> l.contains("WARN"))).isEmpty();
     TestUtils.verifyNoGuiWarnings(ORCHESTRATOR, result);
   }
 
   private SonarScanner getSonarScanner(String projectKey, String projectDir) {
-    return SonarScanner.create(new File(projectDir))
+    File projectDirPath = new File(projectDir);
+    return SonarScanner.create(projectDirPath)
       .setProjectKey(projectKey)
+      .setProperty("sonar.projectBaseDir", projectDirPath.getAbsolutePath())
```
Comment (andrei-epure-sonarsource) [inline its/src/test/java/com/sonar/it/shared/ScannerCliTest.java:113]: @costin-zaharia-sonarsource I would like to get rid of the unrelated WARN messages in the logs - however I set the `sonar.projectBaseDir` and the WARN is still there - is this expected?

Comment (costin-zaharia-sonarsource) [reply its/src/test/java/com/sonar/it/shared/ScannerCliTest.java:113]: We could though get rid of the warning by simulating the scanner behavior and populating `sonar.pullrequest.cache.basepath`.

File: its/src/test/java/com/sonar/it/shared/ScannerCliTest.java [hunk 2]
```diff
@@ -49,11 +49,11 @@ public void givenRazorPagesMainCode_whenScannerForCliIsUsed_logsCSharpWarning()
     BuildResult result = ORCHESTRATOR.executeBuild(scanner);
 
     assertThat(result.getLogsLines(l -> l.contains("WARN")))
-      // ToDo revert back to containsExactlyInAnyOrder https://github.com/SonarSource/sonar-dotnet/issues/6714
-      .contains(
+      .containsExactlyInAnyOrder(
         "WARN: Your project contains C# files which cannot be analyzed with the scanner you are using. To analyze C# or VB.NET, you must use the SonarScanner for .NET 5.x or higher, see https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html",
-        "WARN: Incremental PR analysis: Could not determine common base path, cache will not be computed. Consider setting 'sonar.projectBaseDir' property.",
         "WARN: Your project contains VB.NET files which cannot be analyzed with the scanner you are using. To analyze C# or VB.NET, you must use the SonarScanner for .NET 5.x or higher, see https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html",
+        // the below warnigns are not related to this test; however we still want to be in control of the warning messages
```
Comment (costin-zaharia-sonarsource) [inline its/src/test/java/com/sonar/it/shared/ScannerCliTest.java:55]:
```suggestion
        // the below warnings are not related to this test; however we still want to be in control of the warning messages
```

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! Just a small typo

File: its/src/test/java/com/sonar/it/shared/ScannerCliTest.java (hunk 1 above)
Comment (andrei-epure-sonarsource) [reply its/src/test/java/com/sonar/it/shared/ScannerCliTest.java:116]: I've changed the comments as actually they are related to the usage of the Scanner CLI. Hopefully they are useful

**Comment by costin-zaharia-sonarsource** [comment]
I've took a second look after the comments change and the PR looks good to me.

**Pull request merged**